pkg/deploy/tls.go

```diff
@@ -60,99 +61,141 @@ func IsSelfSignedCertificateUsed(deployContext *DeployContext) (bool, error) {
 		return false, err
 	}
 
-	if util.IsOpenShift {
-		// Get router TLS certificates chain
-		peerCertificates, err := GetEndpointTLSCrtChain(deployContext, "")
-		if err != nil {
-			return false, err
-		}
+	if !util.IsOpenShift {
+		// Handle custom tls secret for Che ingresses
+		cheTLSSecretName := deployContext.CheCluster.Spec.K8s.TlsSecretName
+		if cheTLSSecretName != "" {
+			// The secret is specified in CR
+			cheTLSSecret := &corev1.Secret{}
+			err = deployContext.ClusterAPI.Client.Get(context.TODO(), types.NamespacedName{Namespace: deployContext.CheCluster.Namespace, Name: cheTLSSecretName}, cheTLSSecret)
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					// Failed to get secret, return error to restart reconcile loop.
+					return false, err
+				}
 
-		// Check the chain if ti contains self-signed CA certificate
-		for _, cert := range peerCertificates {
-			if cert.Subject.String() == cert.Issuer.String() {
-				// Self-signed CA certificate is found in the chain
+				// Both secrets (che-tls and self-signed-certificate) are missing which means that we should generate them (i.e. use self-signed certificate).
 				return true, nil
 			}
+			// TLS secret found, consider it as commonly trusted.
```

Comment:
What about the case when self-signed-certificate secret exists but che-tls does not? How do we handle that ?

Reply:
Operator will delete `self-signed-certificate` secret and will generate new pair.